common/utils.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/python
+#
+# Copyright 2022-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+# Common Python APIs and utilities for Intel® Infrastructure Processing Unit (Intel® IPU)
+
+import subprocess, os, time, re
+import yaml
+
+
+def run_cmd(cmd, output=False, check_returncode=True):
+    """
+    Execute a command in a subprocess.
+    :param cmd: The command to be executed as a string.
+    :param output: If True, capture and return the command's stdout.
+    :param check_returncode: If True, check the command's return code and raise an error if not 0.
+    :return: The stdout of the command if output is True, otherwise None.
+    """
+    print(f'Executing: {cmd}')
+    # Use a context manager to ensure the subprocess is cleaned up after execution
+    with subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE if output else None, stderr=subprocess.PIPE, encoding="utf-8") as s:
+        outs, errs = s.communicate() if output else (None, None)
+        # Check the return code if required
+        if check_returncode and s.returncode != 0:
+            raise subprocess.CalledProcessError(s.returncode, cmd, output=outs, stderr=errs)
+        return outs.strip() if output else None
+
+
+def ping_test(dst_ip, count=4, vm = None):
+    if vm:
+        cmd = f"ip netns exec {vm} ping {dst_ip} -c {count}"
+    else:
+        cmd = f"ping {dst_ip} -c {count}"
+    try:
+        result = run_cmd(cmd, output=True)
+        pkt_loss = 100
+        if result:
+            match = re.search('(\d*)% packet loss', result)
+            if match:
+                pkt_loss = int(match.group(1))
+            if f"{count} received, 0% packet loss" in result:
+                print(f"PASS: Ping successful to destination {dst_ip}\n")
+                return True
+            else:
+                raise RuntimeError(f"FAIL: Ping Failed to destination {dst_ip} with" f" {pkt_loss}% loss\n")
+    except Exception as E:
+        print(f"Ping run failed with error:'{E}'\n")
+        return False
+
+
+def split_mac(mac_address):
+    # Split the MAC address by colons
+    octets = mac_address.split(":")
+
+    # Join the first three octets and the last three octets
+    first_octet = "".join(octets[:2])
+    second_octet = "".join(octets[2:4])
+    third_octet = "".join(octets[4:])
+
+    print("first octet {} second octet {} third octet {}".format(first_octet, second_octet, third_octet))
+    return first_octet, second_octet, third_octet
+
+
+def split_mac_2(mac_address):
```

Comment:
Curious why there is a second `split_mac` function.

The second one seems to be just getting the first and second split and returning two values, while the other one splits and returns 3.  If only one is used, the other can be removed

Reply:
Certain P4 tables params takes in 3 MAC split. Example linux_networking_control.rif_mod_table_start needs mac_start, mac_mid and mac_last
Some P4 tables takes in 2 MAC splits : Example linux_networking_control.nexthop_table needs dmac_high and dmac_low

Hence there are two version of split_mac. 
split_mac 
split_mac_2

For IPsec we need both of these